optimum/habana/transformers/models/llama/modeling_llama.py

```diff
@@ -1255,26 +1342,54 @@ def forward(
-                    cache_idx=cache_idx,
-                    num_virtual_tokens=num_virtual_tokens,
-                )
+                if attn_batch_split > 1 and past_key_values is None:
+                    layer_outputs = decoder_layer(
+                        hidden_states,
+                        attention_mask=causal_mask,
+                        position_ids=position_ids,
+                        past_key_value=None if past_key_values is None else past_key_values[layer_idx],
+                        output_attentions=output_attentions,
+                        use_cache=use_cache,
+                        cache_position=cache_position,
+                        position_embeddings=position_embeddings,
+                        token_idx=token_idx,
+                        attn_softmax_bf16=attn_softmax_bf16,
+                        reuse_cache=reuse_cache,
+                        use_flash_attention=use_flash_attention,
+                        flash_attention_recompute=flash_attention_recompute,
+                        flash_attention_causal_mask=flash_attention_causal_mask,
+                        flash_attention_fast_softmax=flash_attention_fast_softmax,
+                        valid_sequence_lengths=valid_sequence_lengths,
+                        cache_idx=cache_idx,
+                        num_virtual_tokens=num_virtual_tokens,
+                        attn_batch_split=attn_batch_split,
+                        prev_layer_residual=prev_layer_residual,
+                    )
+                    index = 1 + int(use_cache) + int(output_attentions)
+                    prev_layer_residual = layer_outputs[index]
+                else:
+                    layer_outputs = decoder_layer(
+                        hidden_states,
+                        attention_mask=causal_mask,
+                        position_ids=position_ids,
+                        past_key_value=None if past_key_values is None else past_key_values[layer_idx],
+                        output_attentions=output_attentions,
+                        use_cache=use_cache,
+                        cache_position=cache_position,
+                        position_embeddings=position_embeddings,
+                        token_idx=token_idx,
+                        attn_softmax_bf16=attn_softmax_bf16,
+                        reuse_cache=reuse_cache,
+                        use_flash_attention=use_flash_attention,
+                        flash_attention_recompute=flash_attention_recompute,
+                        flash_attention_causal_mask=flash_attention_causal_mask,
+                        flash_attention_fast_softmax=flash_attention_fast_softmax,
+                        valid_sequence_lengths=valid_sequence_lengths,
+                        cache_idx=cache_idx,
+                        num_virtual_tokens=num_virtual_tokens,
+                        attn_batch_split=attn_batch_split,
+                    )
+
```

Comment:
The if-else code looks exactly same except, `prev_layer_residual` argument. Could we rewrite it as follows, to avoid code repetition. Later readers/editors of this section will have to undergo some cognitive load figuring out whats the delta between the if-else branches, so if we can can we have only 1 call to decoder_layer?

```
use_prev_layer_residual = attn_batch_split > 1 and past_key_values is None

layer_outputs = decoder_layer(
                        hidden_states,
                        attention_mask=causal_mask,
                        position_ids=position_ids,
                        past_key_value=None if past_key_values is None else past_key_values[layer_idx],
                        output_attentions=output_attentions,
                        use_cache=use_cache,
                        cache_position=cache_position,
                        position_embeddings=position_embeddings,
                        token_idx=token_idx,
                        attn_softmax_bf16=attn_softmax_bf16,
                        reuse_cache=reuse_cache,
                        use_flash_attention=use_flash_attention,
                        flash_attention_recompute=flash_attention_recompute,
                        flash_attention_causal_mask=flash_attention_causal_mask,
                        flash_attention_fast_softmax=flash_attention_fast_softmax,
                        valid_sequence_lengths=valid_sequence_lengths,
                        cache_idx=cache_idx,
                        num_virtual_tokens=num_virtual_tokens,
                        attn_batch_split=attn_batch_split,
                        prev_layer_residual=prev_layer_residual if use_prev_layer_residual  else None,
                    )
                    if use_prev_layer_residual:
                        index = 1 + int(use_cache) + int(output_attentions)
                        prev_layer_residual = layer_outputs[index]
    ```

Reply:
Done